port_definition.go

```diff
@@ -0,0 +1,23 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+// PortDefinition provides metadata about a port
+type PortDefinition struct {
+	Port     int    `json:"port,omitempty"`
```

Comment:
My understanding of the [upstream documentation](https://mesosphere.github.io/marathon/docs/generated/api.html#v2_apps_post) is that the `port` JSON field carries special semantics when set to zero:

> For every port definition with port number zero, a globally unique (cluster-wide) service port is assigned and provided as part of the app definition to be used in load balancing definitions.

This sounds like we must permit to marshal a 0 (zero) value by means of dropping the `omitempty` tag specifier and turn `Port` into a pointer-to-int so that we can represent the three available states missing, zero, and non-zero. We normally implement convenience methods to clear (nilify) and set pointerized variables.

WDYT?